server/src/main/java/org/opensearch/action/admin/cluster/state/term/GetTermVersionResponse.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.admin.cluster.state.term;
+
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.cluster.coordination.ClusterStateTermVersion;
+import org.opensearch.core.action.ActionResponse;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+
+/**
+ * Response object of cluster term
+ *
+ * @opensearch.internal
+ */
+public class GetTermVersionResponse extends ActionResponse {
+
+    private final ClusterStateTermVersion clusterStateTermVersion;
+
+    public GetTermVersionResponse(ClusterStateTermVersion clusterStateTermVersion) {
+        this.clusterStateTermVersion = clusterStateTermVersion;
+    }
+
+    public GetTermVersionResponse(StreamInput in) throws IOException {
+        super(in);
+        this.clusterStateTermVersion = new ClusterStateTermVersion(in);
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        clusterStateTermVersion.writeTo(out);
+    }
+
+    public ClusterStateTermVersion getClusterStateTermVersion() {
+        return clusterStateTermVersion;
+    }
+
+    public boolean matches(ClusterState clusterState) {
+        return clusterStateTermVersion != null
+            && clusterStateTermVersion.getClusterName().equals(clusterState.getClusterName())
+            && clusterStateTermVersion.getClusterUUID().equals(clusterState.metadata().clusterUUID())
+            && clusterStateTermVersion.getTerm() == clusterState.term()
+            && clusterStateTermVersion.getVersion() == clusterState.version();
+    }
+
```

Comment:
Replace with ClusterStateTermVersion 's equals call?

Reply:
The comparison is b/w `clusterState` and `clusterStateTermVersion` and cannot use equals call directly. I can probably move this method into `ClusterStateTermVersion` class if you think there is reusability in future.

Reply:
Can we create a new instances of clusterStateTermVersion from cluster state and then do equals?

Reply:
Created a new constructor to construct clusterStateTermVersion